temporal-sdk/src/test/java/io/temporal/internal/sync/PromiseTest.java

```diff
@@ -55,11 +55,10 @@ public void testFailure() throws Throwable {
             () -> {
               CompletablePromise<Boolean> f = Workflow.newPromise();
               trace.add("root begin");
-              WorkflowInternal.newThread(
-                      false, () -> f.completeExceptionally(new IllegalArgumentException("foo")))
+              WorkflowThread.newThread(
```

Comment:
The idea of WorkflowInternal is to proxy all the public APIs through it and keep all the implementation details private. Making WorkflowThread and its impl public goes against this idea.

Reply:
Yeah, but it's actually not what's going on here.
We had some leftover stuff when internal WorkflowInternal had methods that were just calling WorkflowThread methods.
So, we had an "internal" API that didn't do anything else other than calling a "public" API. So, I just removed these methods and replaced them with using an existing public API.